### PR TITLE
[WIP] ENH: crosstab & crosstabs functions added

### PR DESCRIPTION
**Pull request by skathirmani**
@sanand0 This has few bug fixes to groupmeans too. Ideally I have to submit two merge request. But thought I will just combine them together.

### CODE REVIEW

File: autolysis/__init__.py [hunk 1]
```diff
@@ -210,3 +212,222 @@ def groupmeans(data, groups, numbers,
...
+    ``language`` and ``city`` are independent::
+
+        In [7]: result['V']
+        Out[7]: 0.54866228139712236
+
+    Any value above 0.5 is a strong indication of a relationship. This says
+    that ``language`` and ``city`` are dependent.
+    '''
+    observed = pd.crosstab(index, column, values, aggfunc='sum').fillna(0)
+    chi2, p, dof, expected_values = chi2_contingency(observed.values,
```
Comment (pratapvardhan) [inline autolysis/__init__.py:301]: Shouldn't this func also be Blaze based?

File: autolysis/__init__.py [hunk 2]
```diff
@@ -210,3 +212,222 @@ def groupmeans(data, groups, numbers,
...
+    Some of these insights may be trivial, but all be statistically and
+    numerically significant.
+    '''
+    if columns is None:
+        columns = types(data)['groups']
+
+    result = {}
+    matrix = []
+    parameters = ('p', 'chi2', 'dof', 'V')
+    values = data[values] if values in data else values
```
Comment (pratapvardhan) [inline autolysis/__init__.py]: This is only true for pandas dataframes, not quite for Blaze objects?

File: autolysis/__init__.py [hunk 3]
```diff
@@ -210,3 +212,222 @@ def groupmeans(data, groups, numbers,
...
+
+    Examples
+    --------
+    Consider this example::
+
+        # Load the data
+        >>> from odo import odo
+        >>> import blaze as bz
+        >>> data = odo('http://files.gramener.com/data/trains.csv',
+                       'sqlite:///trains.db::trains')
```
Comment (pratapvardhan) [inline autolysis/__init__.py]: I'm afraid doctests would fail here.

**Comment by pratapvardhan** [comment]
Marking this to WIP version. cc @skathirmani @sanand0

File: autolysis/__init__.py (hunk 1 above)
Comment (skathirmani) [inline autolysis/__init__.py:301]: @pratapvardhan: We have decided it to be of pandas based as of now.

File: autolysis/__init__.py (hunk 2 above)
Comment (skathirmani) [inline autolysis/__init__.py]: @pratapvardhan: True. As of now I haven't considered the parameter values. I should have mentioned these assumptions in the description. Thanks for pointing out.

**Pull request closed without merging**